RingBufCPP.h

```diff
@@ -30,21 +30,65 @@ RingBufCPP()
 */
 bool add(const Type &obj, bool overwrite=false)
 {
-    bool full = false;
+    bool ret = false;
+    
     RB_ATOMIC_START
     {
         full = isFull();
         if (!full || overwrite) {
             _buf[_head] = obj;
             _head = (_head + 1)%MaxElements;
             _numElements = full ? _numElements : (_numElements + 1);
+            ret = true;
         }
     }
     RB_ATOMIC_END
 
-    return !full;
+    return ret;
```

Comment:
This is technically an API change. Now it returns true if an item was inserted versus if the FIFO was full. This might break people who rely on this old behavior. 

Can't the caller just know if overwrite is true then the item will always be inserted?

Reply:
You are right I guess. I misunderstood what to wanted to accomplish with the return value. Since it's an add function, I thought the logical thing to return is if it was successfully added or not.

Reply:
Yeah, I agree this is a bit confusing. Do you want to update the docstring above the method documenting this behavior a bit clearer?